packages/transaction-controller/src/types.ts

```diff
@@ -37,7 +37,7 @@ type TransactionMetaBase = {
   /**
    * Network code as per EIP-155 for this transaction.
    */
-  chainId?: Hex;
+  chainId: Hex;
```

Comment:
Given this was optional in the past, is it conceptually correct to make this required now?

Are we going to add a migration to mobile to populate this on all persisted transactions?

Given both this and `networkID` were optional, can we guarantee all existing persisted transactions have at least one that we could use in a migration?

Reply:
> Given this was optional in the past, is it conceptually correct to make this required now?

I'm unsure. The goal was to tighten the Transaction interfaces. Although they are marked optional, they have been passed in for a long time now. Technically you are right that for mobile that very early transactions may not have `txMeta.chainId` populated but we are marking it required here. 

> Are we going to add a migration to mobile to populate this on all persisted transactions?
> Given both this and networkID were optional, can we guarantee all existing persisted transactions have at least one that we could use in a migration?

At its earliest implementation, `chainId` is guaranteed to exist on `txMeta.txParams` (previously `txMeta.transaction`) and was later added to `txMeta.chainId`. Rather than migrating, we're going to check both locations for now.

Reply:
I see I'm wrong about `txMeta.txParams.chainId being available` due to `normalizeTxParams`

Reply:
We have been populating this field for awhile now and have already essentially been treating it as required. It should be available. Going to leave as required